### PR TITLE
Update docs for deploying to Heroku

### PR DESCRIPTION
**Pull request by marks**
###  Summary

Update Heroku deployment documentation and update the required version of Node to 12.x (Heroku looks at this value when choosing a build pack).

This PR should take care of #25 -- Heroku no longer has a free MongoDB add-on. The documentation now points readers to search for one or use MongoDB Atlas which does currently have a free tier/option.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing guidelines](../CONTRIBUTING.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

**Comment by marks** [comment]
Follow up PR to #30 which addresses issue #25

cc @seratch for when you have a chance

File: docs/DEPLOY_Heroku.md [hunk 1]
```diff
@@ -70,12 +68,32 @@ In a new tab, do the following. Be sure to replace `awesome-app-name-you-entered
     - Under "Subscribe to bot events", add the `app_home_opened` event
     - Click Save Changes
 
-## Part 3: Install and use your app
+## Part 3: Set your MongoDB backend connection information
+
+It's now time to configure your backend, powered by a MongoDB database. The application is set up to look for a [MongoDB connection string](https://docs.mongodb.com/manual/reference/connection-string/) stored in a `MONGODB_URI` [Heroku environment variable](https://devcenter.heroku.com/articles/config-vars).
+
+### Heroku add-ons make it easy
+For the fastest setup, you can look for a Heroku Add-On for MongoDB by browsing the Add-ons marketplace [here](https://elements.heroku.com/addons).
```
Comment (seratch) [inline docs/DEPLOY_Heroku.md:76]: How about filtering by the keyword "MongoDB"?
```suggestion
For the fastest setup, you can look for a Heroku Add-On for MongoDB by browsing the Add-ons marketplace [here](https://elements.heroku.com/search/addons?q=MongoDB).
```

**Review by seratch — APPROVED** [review]
Looks great to me! Thanks a lot for taking the time to update this. I added one minor suggestion about the add-ons search. What do you think?

**Comment by marks** [comment]
Great idea @seratch. Done!

**Pull request merged**